packages/cozy-harvest-lib/src/models/ConnectionFlow.js

```diff
@@ -617,20 +655,45 @@ export class ConnectionFlow {
    * If the trigger we display is already running, subscribe to the associated job
    */
   watchCurrentJobIfTriggerIsAlreadyRunning() {
-    if (get(this, 'trigger.current_state.status') === 'running') {
+    if (this.trigger?.current_state?.status === 'running') {
       this.job = {
-        _id: get(this, 'trigger.current_state.last_executed_job_id')
+        _id: this.trigger?.current_state?.last_executed_job_id
       }
       this.watchJob()
     }
   }
 
+  /**
+   * Watch all jobs related to the current trigger
+   */
+  watchTriggerJobs() {
+    if (this.trigger) {
+      // When the trigger comes from realtime, cozy-stack does not add the current_state to the object. So we need to request the stack to get it
+      if (!this.trigger?.current_state) {
+        this.refetchTrigger()
+      }
+      this.realtime.subscribe(
+        'updated',
+        JOBS_DOCTYPE,
+        this.handleTriggerJobUpdated
+      )
+    }
+  }
+
   watchJob(options = { autoSuccessTimer: false }) {
+    if (this.job?._id === this.jobWatcher?.job?._id) {
+      // no need to rewatch a job we are already watching
+      return
+    }
+    if (this.jobWatcher) {
+      // if no job has been watched yet, there is not jobWatcher yet.
+      this.jobWatcher.unsubscribeAll()
```

Comment:
I still don't understand why we need to `unsubscribeAll()` here?

Reply:
The role of the job watcher is to watch a specific job with a specified id. Since in this case the id of the job to monitor has changed, we tell to  je job watcher to unsubscribe from events associated to the previous job id.